### PR TITLE
Some unit test warnings addressed.

### PR DESCRIPTION
**Pull request by CTLaChance**
This pull request has taken care of all unit test warnings as well as a few compilation warnings.

The issue referenced by this pull request is: #504.

### CODE REVIEW

File: trick_source/sim_services/JSONVariableServer/JSONVariableServer.cpp [hunk 1]
```diff
@@ -25,7 +25,7 @@ Trick::JSONVariableServer::JSONVariableServer() :
 
 Trick::JSONVariableServer::~JSONVariableServer() {
     if ( listen_dev.hostname != NULL ) {
-        free(listen_dev.hostname) ;
+        free((char*)listen_dev.hostname) ;
```
Comment (dbankieris) [inline trick_source/sim_services/JSONVariableServer/JSONVariableServer.cpp]: Why is this necessary? `listen_dev.hostname` is already a `char*`.

File: trick_source/sim_services/VariableServer/VariableServerListenThread.cpp [hunk 2]
```diff
@@ -27,7 +27,7 @@ Trick::VariableServerListenThread::VariableServerListenThread() :
 
 Trick::VariableServerListenThread::~VariableServerListenThread() {
     if ( listen_dev.hostname != NULL ) {
```
Comment (dbankieris) [inline trick_source/sim_services/VariableServer/VariableServerListenThread.cpp]: Might as well fix this while you're in here. No need to check for null pointers before freeing.
https://stackoverflow.com/a/6084233

File: trick_source/trick_utils/comm/test/TCConnectTest.cpp [hunk 3]
```diff
@@ -20,7 +20,7 @@ class TCConnectTest : public testing::Test {
          /* device */
          device = (TCDevice *) malloc(sizeof(TCDevice));
          memset( (void *)device,'\0',sizeof(TCDevice) );
-         device->hostname = "127.0.0.1";
+         device->hostname = (char*)"127.0.0.1";
```
Comment (dbankieris) [inline trick_source/trick_utils/comm/test/TCConnectTest.cpp]: This is [not safe](https://arduino.stackexchange.com/a/13443). We should really make `hostname` a `const char*` in `TCDevice`. Barring that, we should use `strdup` here to make a truly mutable string.

File: trick_source/trick_utils/comm/test/TCConnectTest.cpp [hunk 4]
```diff
@@ -20,7 +20,7 @@ class TCConnectTest : public testing::Test {
          /* device */
          device = (TCDevice *) malloc(sizeof(TCDevice));
          memset( (void *)device,'\0',sizeof(TCDevice) );
```
Comment (dbankieris) [inline trick_source/trick_utils/comm/test/TCConnectTest.cpp]: Another opportunity to fix someone else's code! The argument to `sizeof` should be the [variable itself](https://softwareengineering.stackexchange.com/questions/201104/sizeof-style-sizeoftype-or-sizeof-variable), when possible, not its type. That way, if the type changes in the future, this line doesn't need to be changed.

File: trick_source/sim_services/JSONVariableServer/JSONVariableServer.cpp (hunk 1 above)
Comment (CTLaChance) [reply trick_source/sim_services/JSONVariableServer/JSONVariableServer.cpp]: The free or the cast to a char*?

File: trick_source/sim_services/VariableServer/VariableServerListenThread.cpp (hunk 2 above)
Comment (CTLaChance) [reply trick_source/sim_services/VariableServer/VariableServerListenThread.cpp]: Got it.

File: trick_source/trick_utils/comm/test/TCConnectTest.cpp (hunk 4 above)
Comment (CTLaChance) [reply trick_source/trick_utils/comm/test/TCConnectTest.cpp]: Okey poke. Will fix.

File: trick_source/sim_services/JSONVariableServer/JSONVariableServer.cpp (hunk 1 above)
Comment (CTLaChance) [reply trick_source/sim_services/JSONVariableServer/JSONVariableServer.cpp]: Good catch. I have no idea why I initially did that. I'll fix it.

Comment (CTLaChance) [reply trick_source/sim_services/JSONVariableServer/JSONVariableServer.cpp]: I remember now! I did that because I changed listen_dev.hostname to a const char* and it was throwing an error when I tried to free a constant.

File: trick_source/trick_utils/comm/test/TCConnectTest.cpp (hunk 3 above)
Comment (CTLaChance) [reply trick_source/trick_utils/comm/test/TCConnectTest.cpp]: Hostname unfortunately can't be a const. I've tried it and const is discarded down the line when hostname is reassigned. I'll try using strdup and duplicate the literal into the char *.
